### PR TITLE
Update tests checking launch of workspace from gitlab repository with a .devfile.yaml

### PR DESCRIPTION
**Pull request by artaleks9**
### What does this PR do?
Adds checking launch of workspace from gitlab repository that has .devfile.yaml

### What issues does this PR fix or reference?

https://github.com/eclipse-che/che/issues/23255

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [ ] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [ ] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: .ci/openshift-ci/test-gitlab-with-pat-setup-flow.sh [hunk 1]
```diff
@@ -25,6 +25,12 @@ export GIT_PROVIDER_URL=${GIT_PROVIDER_URL:-"https://gitlab.com"}
 export PRIVATE_REPO_SSH_URL=${PRIVATE_REPO_SSH_URL:-"git@gitlab.com:chepullreq1/private-repo.git"}
 export PRIVATE_REPO_RAW_PATH_URL=${PRIVATE_REPO_URL:-"https://gitlab.com/chepullreq1/private-repo/-/raw/main/devfile.yaml"}
 
+export PUBLIC_REPO_DOT_DEVFILE_URL=${PUBLIC_REPO_DOT_DEVFILE_URL:-"https://gitlab.com/chepullreq1/public-repo-dot-devfile.git"}
```
Comment (dmytro-ndp) [inline .ci/openshift-ci/test-gitlab-with-pat-setup-flow.sh:28]: Variable name is a little bit confusing to me.

Did you mean something like `PUBLIC_REPO_WITH_DOT_DEFILE_URL`?

File: .ci/openshift-ci/test-gitlab-with-pat-setup-flow.sh [hunk 2]
```diff
@@ -25,6 +25,12 @@ export GIT_PROVIDER_URL=${GIT_PROVIDER_URL:-"https://gitlab.com"}
...
+export PRIVATE_REPO_DOT_DEVFILE_URL=${PRIVATE_REPO_DOT_DEVFILE_URL:-"https://gitlab.com/chepullreq1/private-repo-dot-devfile.git"}
```
Comment (dmytro-ndp) [inline .ci/openshift-ci/test-gitlab-with-pat-setup-flow.sh:29]: Variable name is a little bit confusing to me.

Did you mean something like `PRIVATE_REPO_WITH_DOT_DEFILE_URL`?

File: .ci/openshift-ci/test-gitlab-with-pat-setup-flow.sh [hunk 3]
```diff
@@ -25,6 +25,12 @@ export GIT_PROVIDER_URL=${GIT_PROVIDER_URL:-"https://gitlab.com"}
...
+
+export PUBLIC_PROJECT_DOT_DEVFILE_NAME=${PUBLIC_PROJECT_DOT_DEVFILE_NAME:-"public-repo-dot-devfile"}
```
Comment (dmytro-ndp) [inline .ci/openshift-ci/test-gitlab-with-pat-setup-flow.sh:31]: Variable name is a little bit confusing to me.

Did you mean something like `NAME_OF_PUBLIC_REPO_WITH_DOT_DEFILE`?

File: .ci/openshift-ci/test-gitlab-with-pat-setup-flow.sh [hunk 4]
```diff
@@ -25,6 +25,12 @@ export GIT_PROVIDER_URL=${GIT_PROVIDER_URL:-"https://gitlab.com"}
...
+export PRIVATE_PROJECT_DOT_DEVFILE_NAME=${PRIVATE_PROJECT_DOT_DEVFILE_NAME:-"private-repo-dot-devfile"}
```
Comment (dmytro-ndp) [inline .ci/openshift-ci/test-gitlab-with-pat-setup-flow.sh:32]: Variable name is a little bit confusing to me.

Did you mean something like `NAME_OF_PRIVATE_REPO_WITH_DOT_DEFILE`?

File: .ci/openshift-ci/common.sh [hunk 5]
```diff
@@ -555,12 +555,12 @@ testCloneGitRepoNoProjectExists() {
...
     testProjectIsCloned ${PROJECT_NAME} ${OCP_USER_NAMESPACE} && \
-    { echo "####### [ERROR] Project file /projects/${PROJECT_NAME}/${YAML_FILE_NAME} should NOT be present. #######
+    { echo "####### [ERROR] Project file /projects/${PROJECT_NAME}/${TEST_FILE_NAME} should NOT be present. #######
 ####### Cause possible: PR code regress or service is changed. Need to investigate it. #######" && exit 1; }
-    echo "======= [INFO] Project file /projects/${PROJECT_NAME}/${YAML_FILE_NAME} is NOT present. This is EXPECTED. ======="
+    echo "======= [INFO] Project file /projects/${PROJECT_NAME}/${TEST_FILE_NAME} is NOT present. This is EXPECTED. ======="
 }
 
-# Test that the repository is cloned when PAT, OAuth or SSH is configured
+# Test that the repository is cloned when a public repositoty, PAT, OAuth or SSH is configured
```
Comment (dmytro-ndp) [inline .ci/openshift-ci/common.sh:563]: It's not clear what `... the repository is cloned when a public repository.... is configured` means.

**Review by dmytro-ndp — APPROVED** [review]

File: .ci/openshift-ci/common.sh (hunk 5 above)
Comment (artaleks9) [reply .ci/openshift-ci/common.sh]: The annotation is fixed.

**Review by vinokurig — APPROVED** [review]

File: .ci/openshift-ci/test-gitlab-with-pat-setup-flow.sh (hunk 1 above)
Comment (artaleks9) [reply .ci/openshift-ci/test-gitlab-with-pat-setup-flow.sh]: Fixed.

File: .ci/openshift-ci/test-gitlab-with-pat-setup-flow.sh (hunk 3 above)
Comment (artaleks9) [reply .ci/openshift-ci/test-gitlab-with-pat-setup-flow.sh]: Fixed.

File: .ci/openshift-ci/test-gitlab-with-pat-setup-flow.sh (hunk 4 above)
Comment (artaleks9) [reply .ci/openshift-ci/test-gitlab-with-pat-setup-flow.sh]: Fixed.

**Comment by devspacesbuild** [comment]
Build [3.18](https://github.com/redhat-developer/devspaces-images/tree/devspaces-3-rhel-8/) :: [server_3.x/374](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/server_3.x/374/): [Console](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/server_3.x/374/console), [Changes](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/server_3.x/374/changes), [Git Data](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/server_3.x/374/git)

**Comment by devspacesbuild** [comment]
Build [3.18](https://github.com/redhat-developer/devspaces-images/tree/devspaces-3-rhel-8/) :: [sync-to-downstream_3.x/8121](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/sync-to-downstream_3.x/8121/): [Console](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/sync-to-downstream_3.x/8121/console), [Changes](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/sync-to-downstream_3.x/8121/changes), [Git Data](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/sync-to-downstream_3.x/8121/git)

**Comment by devspacesbuild** [comment]
Build [3.18](https://github.com/redhat-developer/devspaces-images/tree/devspaces-3-rhel-8/) :: [get-sources-rhpkg-container-build_3.x/8198](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/get-sources-rhpkg-container-build_3.x/8198/): [Console](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/get-sources-rhpkg-container-build_3.x/8198/console), [Changes](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/get-sources-rhpkg-container-build_3.x/8198/changes), [Git Data](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/get-sources-rhpkg-container-build_3.x/8198/git)

**Comment by devspacesbuild** [comment]
Build [3.18](https://github.com/redhat-developer/devspaces-images/tree/devspaces-3-rhel-8/) :: [push-latest-container-to-quay_3.x/5026](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/push-latest-container-to-quay_3.x/5026/): [Console](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/push-latest-container-to-quay_3.x/5026/console), [Changes](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/push-latest-container-to-quay_3.x/5026/changes), [Git Data](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/push-latest-container-to-quay_3.x/5026/git)

**Comment by devspacesbuild** [comment]
Build [3.18](https://github.com/redhat-developer/devspaces-images/tree/devspaces-3-rhel-8/) :: [get-sources-rhpkg-container-build_3.x/8198](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/get-sources-rhpkg-container-build_3.x/8198/): [SUCCESS](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/get-sources-rhpkg-container-build_3.x/8198/)<br /><blockquote>server : 3.x :: <a href='https://brewweb.engineering.redhat.com/brew/taskinfo?taskID=65969802'> Build  65969802</a> : <a href=https://quay.io/devspaces/server-rhel8:3.18-13>quay.io/devspaces/server-rhel8:3.18-13</a> <br /> SUCCESS; copied to quay; <a href=https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/push-latest-container-to-quay_3.x/5026/>/DS_CI/push-latest-container-to-quay_3.x/5026</a> triggered</blockquote>

**Comment by devspacesbuild** [comment]
Build [3.18](https://github.com/redhat-developer/devspaces-images/tree/devspaces-3-rhel-8/) :: [update-digests_3.x/8020](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/update-digests_3.x/8020/): [Console](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/update-digests_3.x/8020/console), [Changes](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/update-digests_3.x/8020/changes), [Git Data](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/update-digests_3.x/8020/git)

**Comment by devspacesbuild** [comment]
Build [3.18](https://github.com/redhat-developer/devspaces-images/tree/devspaces-3-rhel-8/) :: [server_3.x/374](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/server_3.x/374/): [SUCCESS](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/server_3.x/374/)<br /><blockquote>Upstream sync done; <a href=https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/sync-to-downstream_3.x/8121/>/DS_CI/sync-to-downstream_3.x/8121</a> triggered</blockquote>

**Comment by devspacesbuild** [comment]
Build [3.18](https://github.com/redhat-developer/devspaces-images/tree/devspaces-3-rhel-8/) :: [get-sources-rhpkg-container-build_3.x/8200](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/get-sources-rhpkg-container-build_3.x/8200/): [Console](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/get-sources-rhpkg-container-build_3.x/8200/console), [Changes](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/get-sources-rhpkg-container-build_3.x/8200/changes), [Git Data](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/get-sources-rhpkg-container-build_3.x/8200/git)

**Comment by devspacesbuild** [comment]
Build [3.18](https://github.com/redhat-developer/devspaces-images/tree/devspaces-3-rhel-8/) :: [get-sources-rhpkg-container-build_3.x/8200](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/get-sources-rhpkg-container-build_3.x/8200/): [FAILURE](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/DS_CI/job/get-sources-rhpkg-container-build_3.x/8200/)<br /><blockquote>devspaces-operator-bundle : 3.x :: <a href='https://brewweb.engineering.redhat.com/brew/taskinfo?taskID=65970450'> Failed in  65970450</a> : BREW:BUILD/STATUS:UNKNOWN <br /> FAILURE:; copied to quay</blockquote>

**Pull request merged**